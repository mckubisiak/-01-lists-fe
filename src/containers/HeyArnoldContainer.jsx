import React, { Component } from 'react';
import Characterlist from '../components/app/Characterlist';
import { fetchCharacters } from '../services/heyArnoldApi';

export default class HeyArnoldContainer extends Component {
  state = {
    loading: true,
    characters: [],
  };

  componentDidMount() {
    fetchCharacters().then((characters) =>
      this.setState({ loading: false, characters })
    );
  }

  render() {
    if (this.state.loading) {
      return <p>Loadin...!</p>;
    }
    return <Characterlist characters={this.state.characters} />;
  }
}
