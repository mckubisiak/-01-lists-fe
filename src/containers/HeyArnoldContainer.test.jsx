import React from 'react';
import { screen, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import HeyArnoldContainer from './HeyArnoldContainer';

describe('HeyArnoldContainer', () => {
  it('renders lists of characters', async () => {
    render(
      <MemoryRouter>
        <HeyArnoldContainer />
      </MemoryRouter>
    );

    screen.getByText('Loadin...!');

    const ul = await screen.findByRole('list', { name: 'characters' });

    console.log(ul);

    expect(ul).toMatchSnapshot();
  });
});
