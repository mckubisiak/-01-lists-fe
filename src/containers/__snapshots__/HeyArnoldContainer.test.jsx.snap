// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeyArnoldContainer renders lists of characters 1`] = `
<ul
  aria-label="characters"
  name="characters"
  role="list"
>
  <li>
    <a
      href="/5da237699734fdcb7bef8f51"
    >
      Arnold Shortman
    </a>
    <img
      alt="5da237699734fdcb7bef8f51"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f52"
    >
      Grandpa Phil
    </a>
    <img
      alt="5da237699734fdcb7bef8f52"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f53"
    >
      Grandma Gertie
    </a>
    <img
      alt="5da237699734fdcb7bef8f53"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/7/7b/DB87AC2A-C2C2-4DAB-AECF-B9F7273C746B.png/revision/latest/scale-to-width-down/306?cb=20190816101257"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f54"
    >
      Mitzi
    </a>
    <img
      alt="5da237699734fdcb7bef8f54"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f55"
    >
      Arnie
    </a>
    <img
      alt="5da237699734fdcb7bef8f55"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f56"
    >
      Miles Shortman
    </a>
    <img
      alt="5da237699734fdcb7bef8f56"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f57"
    >
      Stella Shortman
    </a>
    <img
      alt="5da237699734fdcb7bef8f57"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fa/Stella.jpg/revision/latest?cb=20101225143956"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f58"
    >
      Grandpa's father
    </a>
    <img
      alt="5da237699734fdcb7bef8f58"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f59"
    >
      Grandpa's grandfather
    </a>
    <img
      alt="5da237699734fdcb7bef8f59"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5a"
    >
      Oskar Kokoshka
    </a>
    <img
      alt="5da237699734fdcb7bef8f5a"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/c/c5/Oskar_Kokoshka.png/revision/latest/scale-to-width-down/310?cb=20180707032418"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5b"
    >
      Suzie Kokoshka
    </a>
    <img
      alt="5da237699734fdcb7bef8f5b"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fe/Suzie.jpg/revision/latest/scale-to-width-down/310?cb=20110223003447"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5c"
    >
      Nancy
    </a>
    <img
      alt="5da237699734fdcb7bef8f5c"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5d"
    >
      Ernie Potts
    </a>
    <img
      alt="5da237699734fdcb7bef8f5d"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/6/67/Ernie.jpeg/revision/latest/scale-to-width-down/310?cb=20181213181427"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5e"
    >
      Mr. Hyunh
    </a>
    <img
      alt="5da237699734fdcb7bef8f5e"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/6/60/Mr._Hyunh.png/revision/latest/scale-to-width-down/310?cb=20180707031441"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f5f"
    >
      Mai Hyunh
    </a>
    <img
      alt="5da237699734fdcb7bef8f5f"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/8/83/Mai_Hyunh.jpeg/revision/latest/scale-to-width-down/310?cb=20190118192457"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f60"
    >
      Mr. Smith
    </a>
    <img
      alt="5da237699734fdcb7bef8f60"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/8/80/Mr._Smith.jpg/revision/latest?cb=20110206233402"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f61"
    >
      Lana Vail
    </a>
    <img
      alt="5da237699734fdcb7bef8f61"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/762C69BE-23BA-48BA-9B14-6AF3246FCD3F.jpeg/revision/latest/scale-to-width-down/304?cb=20190106211607"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f62"
    >
      Mr. Purdy (character)
    </a>
    <img
      alt="5da237699734fdcb7bef8f62"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/9/9c/Mr._Purdy_%28character%29.jpeg/revision/latest?cb=20190125214742"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f63"
    >
      Helga G. Pataki
    </a>
    <img
      alt="5da237699734fdcb7bef8f63"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807"
    />
  </li>
  <li>
    <a
      href="/5da237699734fdcb7bef8f64"
    >
      Robert "Big Bob" Pataki
    </a>
    <img
      alt="5da237699734fdcb7bef8f64"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Big_Bob.jpg/revision/latest/scale-to-width-down/310?cb=20170428021615"
    />
  </li>
</ul>
`;
